searchbase/lib/src/searchcontroller.dart

```diff
@@ -922,8 +922,10 @@ class SearchController extends Base {
   /// -   `customQuery`
   ///
   subscribeToStateChanges(
-      SubscriptionFunction fn, List<String>? propertiesToSubscribe) {
-    this.stateChanges.subscribe(fn, propertiesToSubscribe);
+      SubscriptionFunction fn, List<KeysToSubscribe>? propertiesToSubscribe) {
+
+    final keysToSubscribeValues = propertiesToSubscribe?.map((item)=>item.name).toList();
```

Comment:
@mohdashraf010897 What are you doing here?
`stateChanges.subscribe` should accept the `KeysToSubscribe` type.